ZigbeeNet/CC/CCZnp.cs

```diff
@@ -262,31 +128,32 @@ private void Unpi_DataReceived(object sender, SerialPacket e)
 
                     return;
                 }
-                ZpiObject zpiObject = new ZpiObject((SubSystem)e.SubSystem, (MessageType)e.Type, e.Cmd1);
-                zpiObject.OnParsed += (object s, ZpiObject result) =>
+
+                var zpiObject = new ZpiObject(e.SubSystem, e.Type, e.Cmd1);
+                zpiObject.OnParsed += (s, result) =>
                 {
                     _logger.Info("{Type} - {SubSystem} - {Name}", e.Type, result.SubSystem, result.Name);
 
                     AsyncResponse?.Invoke(this, result);
                 };
-                zpiObject.Parse((MessageType)e.Type, e.Length, e.Payload);
+                zpiObject.Parse(e.Type, e.Length, e.Payload);
             }
-            else if ((byte)e.Type == (byte)MessageType.SRSP)
+            else if ((byte) e.Type == (byte) MessageType.SRSP)
             {
                 if (_sreqRunning != null)
-                {
-                    if ((byte)_sreqRunning.SubSystem == (byte)e.SubSystem && _sreqRunning.CommandId == e.Cmd1)
+                    if ((byte) _sreqRunning.SubSystem == (byte) e.SubSystem && _sreqRunning.CommandId == e.Cmd1)
                     {
                         // Status Response
                         _timeout.Change(Timeout.Infinite, Timeout.Infinite);
 
                         _resetting = false;
 
-                        _sreqRunning.OnParsed += (object s, ZpiObject result) =>
+                        _sreqRunning.OnParsed += (s, result) =>
                         {
-                            ZpiSREQ sREQ = (ZpiSREQ)result;
+                            var sREQ = (ZpiSREQ) result;
```

Comment:
Why do you use `var `instead of `ZpiSREQ `if type is clear by declaration?

Reply:
If declaration was unclear, it would be a compile error. I always use var, as I use auto in C++.

Reply:
Yes, you have to use a declaration type, but i think it is easier to read Code if you write int instead of var, or ZpiSREQ instead var. Sometimes you create a variable out of combining other or do calculation. Then the correct data type would be better then var